libs/wingsdk/src/shared-gcp/bucket.inflight.ts

```diff
@@ -0,0 +1,257 @@
+import { Storage, Bucket } from "@google-cloud/storage";
+import {
+  BucketDeleteOptions,
+  IBucketClient,
+  ObjectMetadata,
+  BucketSignedUrlOptions,
+} from "../cloud";
+import { Json } from "../std";
+
+export class BucketClient implements IBucketClient {
+  private bucketName: string;
+  private storage: Storage;
+  private bucket: Bucket;
+
+  constructor(
+    bucketName: string,
+    storage: Storage,
+    projectId: string = process.env.GCP_PROJECT_ID || "wingsdk-test"
+  ) {
+    this.bucketName = bucketName;
+    this.storage = storage ? storage : new Storage({ projectId: projectId });
```

Comment:
How come there is a default initialization? Based on the constructor's type annotation (`storage: Storage`) it looks like it should always be defined

Reply:
So should I only do this?
```js
    this.storage = storage;
```